memphis/station.go

```diff
@@ -0,0 +1,224 @@
+package memphis
+
+import "github.com/nats-io/nats.go"
+
+type Station struct {
+	Name              string
+	RetentionType     RetentionType
+	RetentionValue    int
+	StorageType       StorageType
+	Replicas          int
+	DedupEnabled      bool
+	DedupWindowMillis int
+	factoryName       string
+	conn              *Conn
+}
+
+type RetentionType int
+
+const (
+	MaxMessageAgeSeconds RetentionType = iota
+	Messages
+	Bytes
+)
+
+func (r RetentionType) String() string {
+	return [...]string{"message_age_sec", "messages", "bytes"}[r]
+}
+
+type StorageType int
+
+const (
+	File StorageType = iota
+	Memory
+)
+
+func (s StorageType) String() string {
+	return [...]string{"file", "memory"}[s]
+}
+
+type CreateStationReq struct {
+	Name              string `json:"name"`
+	FactoryName       string `json:"factory_name"`
+	RetentionType     string `json:"retention_type"`
+	RetentionValue    int    `json:"retention_value"`
+	StorageType       string `json:"storage_type"`
+	Replicas          int    `json:"replicas"`
+	DedupEnabled      bool   `json:"dedup_enabled"`
+	DedupWindowMillis int    `json:"dedup_window_in_ms"`
+}
+
+type RemoveStationReq struct {
+	Name string `json:"station_name"`
+}
+
+type GetStationReq struct {
+	Name string `json:"station_name"`
+}
+type StationOpts struct {
+	Name              string
+	FactoryName       string
+	RetentionType     RetentionType
+	RetentionVal      int
+	StorageType       StorageType
+	Replicas          int
+	DedupEnabled      bool
+	DedupWindowMillis int
+}
+
+type StationOpt func(*StationOpts) error
+
+func GetStationDefaultOptions() StationOpts {
```

Comment:
that is actually useful in case the user wants to use the stationOptions.CreateStation() 
a user can use this to get the default values struct and alter the fields instead of putting a list of parameter functions to the connection.CreateStation(...here...) call.

or start with an empty struct and fill all fields.. could be risky though